com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/visitor/CobolVisitor.java

```diff
@@ -138,13 +167,20 @@ public Class visitDataDescriptionEntryFormat1(DataDescriptionEntryFormat1Context
     if (tokenText.equals("01") || tokenText.equals("1") || tokenText.equals("77")) {
       areaAWarning(ctx.start.getCharPositionInLine(), tokenText, token.getLine());
     }
+    String name = Optional.ofNullable(ctx.dataName1()).map(RuleContext::getText).orElse("FILLER");
+    int level = Integer.parseInt(ctx.otherLevel().getText());
+    boolean hasPic = !ctx.dataPictureClause().isEmpty();
+    boolean hasRedefines = !ctx.dataRedefinesClause().isEmpty();
+    NodeType nodeType = hasPic ? NodeType.FIELD : hasRedefines ? NodeType.REDEFINES : NodeType.STRUCT;
```

Comment:
optional: it's super cool but I am afraid that will be a bit complicated for the developer understand what is going on here,
maybe extract the last  part (``` hasRedefines ? NodeType.REDEFINES : NodeType.STRUCT```) may help

Reply:
I tried to improve this part by comments. Is it good now?